### PR TITLE
feat: Strip username from uploaded logs

### LINKED ISSUE #405: [Feature Request] Anonymize user path in uploaded log
**Is your feature request related to a problem? Please describe.**
Users are sometimes worried about compromising their privacy when using the "Upload Log" feature.

**Describe the solution you'd like**
Usernames should be automatically removed from paths in the uploaded logs. For example, `C:\Users\nosamu` would become `C:\Users\***`. This could easily be done in a postprocessing step by searching for the value of %USERPROFILE% within the log and replacing the value of %USERNAME% with `***`. Similar replacements should be possible on Linux and Mac too.

**Describe alternatives you've considered**
The feature could instead be implemented server-side on logs.unstable.life, but I think it's better not to store the data at all.

### PR DESCRIPTION
**Pull request by oblivioncth**
Closes #405.

### CODE REVIEW

**Comment by oblivioncth** [comment]
Debating on if this should be more sophisticated, perhaps only replacing the username if it falls between forward/backslashes to avoid false positives in case a username is similar to a common word.

**Comment by n0samu** [comment]
> perhaps only replacing the username if it falls between forward/backslashes to avoid false positives

That sounds like a good idea to me!

File: src/back/responses.ts [hunk 1]
```diff
@@ -1780,8 +1780,14 @@ export function registerRequestCallbacks(state: BackState, init: () => Promise<v
   });
 
   state.socketServer.register(BackIn.UPLOAD_LOG, async (event) => {
+    // Filter entries
+    const username = os.userInfo().username;
+    const entries = state.log.filter(e => e !== undefined).map(e => {
+      e.content = e.content.replace(new RegExp(username, 'g'), '***');
```
Comment (n0samu) [inline src/back/responses.ts:1786]: Try this:
```ts
const slashMatch = '([\/\\\\])'
e.content = e.content.replace(new RegExp(slashMatch + username + slashMatch, 'g'), '$1***$2');
```

Comment (oblivioncth) [reply src/back/responses.ts]: LGTM. Made the second slash optional for the rare case that the path ends with the user folder. Technically allows for a few fringe false positives, but I think that's alright.

**Comment by oblivioncth** [comment]
[Example](https://www.typescriptlang.org/play?#code/MYewdgzgLgBArhApgJzAQwLaJgXhgcgCkBLDDAT3wG4AoUSWYAC0WAGsIAuGaZYsAOYBtALq4YQmjGkwA9LJgBlJiDgAbACYxkiAA5q0wRFJn4AwpwA6lgKpJkEayTKUANCen4wIWADNVYFr8MBbWdiiOls4U1rpo0Gj47jIEobb2kdHk1gCyIDrWilBwvr5JHqlW6RFOpBQwAO75GuUp+LIqWLJZraYdIF1Zsnk6skUlZcl9nYjddeSNzeUV8koq6hpg+LA6+obGbVmN8fBILVOeQwAqiFoNJyw6yymryqqaWzt6BkauMBogRAQP4AIzgsGIEE+MDQMBBxAE-0QaDUFXwNw0cxcxwgMEeiHwNBEtHo0B4UGQAH1kAAPcT4AAUQlk1msIgAlITSTs6XgwIgGjAAEqIAQAURpugZvGpdIA1KcUOgsDAFTLaaqCAB+JIEAT4dkk8Bk3Y-W7iZisDgAOgwaCluJwAD4eNbTfsGbS-vgACQARgAVEGfQAmA3sujGkBqRDWtQgASe777DSGoA)

File: src/back/responses.ts (hunk 1 above)
Comment (n0samu) [reply src/back/responses.ts]: If the second slash is optional, then I'm pretty sure there's no reason to include it in the regex, and you could get an equivalent result by just matching `sepMatch + username` and replacing with `$1***`.

Comment (oblivioncth) [reply src/back/responses.ts]: Of course... because there's no more to the pattern after that. Woops XD.

Revised.

**Review by n0samu — APPROVED** [review]

**Pull request merged**